server/src/test/java/io/druid/curator/announcement/AnnouncerTest.java

```diff
@@ -231,4 +231,29 @@ public void testLeavesBehindTurdlingsThatAlreadyExisted() throws Exception
     Assert.assertEquals(initialStat.getMzxid(), curator.checkExists().forPath(parent).getMzxid());
   }
 
+  @Test
+  public void testLeavesBehindTurdlingsWhenToldTo() throws Exception
+  {
+    curator.start();
+    Announcer announcer = new Announcer(curator, exec);
+
+    final byte[] billy = "billy".getBytes();
+    final String testPath = "/somewhere/test2";
+    final String parent = ZKPaths.getPathAndNode(testPath).getPath();
+
+    announcer.start();
+    try {
+      Assert.assertNull(curator.checkExists().forPath(parent));
+
+      announcer.announce(testPath, billy, false);
```

Comment:
this is not synchronous. Can you have some means of retrying until you are sure curator is caught up?